include/ique/PR/os_cont.h

```diff
@@ -46,42 +49,377 @@ extern "C" {
  *
  */
 
+// -- Analog pairs --
+
+typedef union {
+    struct PACKED { s8 x:4, y:4; };
+    struct PACKED { s8 l:4, r:4; };
+    struct PACKED { s8 a:4, b:4; };
+    s8 raw;
+} Analog_s4; /*0x01*/
+typedef union {
+    struct PACKED { u8 x:4, y:4; };
+    struct PACKED { u8 l:4, r:4; };
+    struct PACKED { u8 a:4, b:4; };
+    u8 raw;
+} Analog_u4; /*0x01*/
+
+typedef union {
+    struct PACKED { s8 x:8, y:8; };
+    struct PACKED { s8 l:8, r:8; };
+    struct PACKED { s8 a:8, b:8; };
+    s16 raw;
+} Analog_s8; /*0x02*/
+typedef union {
+    struct PACKED { u8 x:8, y:8; };
+    struct PACKED { u8 l:8, r:8; };
+    struct PACKED { u8 a:8, b:8; };
+    u16 raw;
+} Analog_u8; /*0x02*/
+
+#define ANALOG_U4_TO_U8(src) ((Analog_u8){  \
+    ((src).a << 4),                         \
+    ((src).b << 4),                         \
+})
+
+//! TODO: CLAMP_S8 is from math_util.h. Should it be included in this file?
```

Comment:
you might wanna remove these comments, also maybe wait until the math_util pr is merged but yeah just include that file

Reply:
I moved these macros to joybus.h instead.